### DISCUSSION TITLE
/date-range-filter-date-picker-display-template/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#106] [Comments]
Date range filter with date picker display template
        
[https://www.eliostruyf.com/date-range-filter-date-picker-display-template/](https://www.eliostruyf.com/date-range-filter-date-picker-display-template/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Mary on 2015-04-12 18:20
```

Hi,

I was really excited when I found your post. Thanks  
However, when I am trying to use it on mys site, no date picker is displayed. Any suggestions?

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-04-13 20:51
```

Mary,

Is jQuery by any chance already loaded in your environment or search page?

Regards,  
Elio

**Reply by comment-migrator** [reply]
```header
Originally posted by Mary on 2015-04-14 13:44
```

Yes. I've made another customization and had to load some jQuery on page layout level.

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-04-14 14:34
```

Mary,

Try to remove the following lines:

RegisterSod('jquery-1.11.0.min.js', "//code.jquery.com/jquery-1.11.0.min.js");

RegisterSodDep('jquery-ui.min.js', 'jquery-1.11.0.min.js');

Let me know if it helps.

Regards,  
Elio

**Comment by comment-migrator** [comment]
```header
Originally posted by Mary on 2015-04-14 17:34
```

Sorry, it didn't

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-04-14 18:14
```

Are you getting any errors in the developer console of your browser?

**Comment by comment-migrator** [comment]
```header
Originally posted by S on 2015-07-28 08:15
```

Hello, you also disappears refiner control if it does not find anything for that date?

**Reply by comment-migrator** [reply]
```header
Originally posted by Jessie on 2016-04-05 02:44
```

I have the same issue. The control disappears when no result is returned. Did you get it resolved? Thanks

**Reply by comment-migrator** [reply]
```header
Originally posted on 2016-04-05 16:51
```

That is how refiners work, if there is no data to return, then there are no refiners which will get loaded.

You could create a link to reset all the refiners that are set from the control refiner template. Here is a template which helps you achieve this: [https://github.com/SPCSR/Di...](https://github.com/SPCSR/DisplayTemplates/tree/master/Search%20Display%20Templates/Refiners/Clear%20all%20refiners%20control%20template "https://github.com/SPCSR/DisplayTemplates/tree/master/Search%20Display%20Templates/Refiners/Clear%20all%20refiners%20control%20template")

**Comment by comment-migrator** [comment]
```header
Originally posted by Pablo Cafferata on 2015-09-02 16:31
```

Your posts are great, Thanks for sharing your knowledge. I want ask you 2 questions:

1.- How can I define values as default for each refiner control, for example today date to datepicker, etc?

2.- How can I refine available dates using others refiner control? with that approach I could only select valid dates.

Thanks in advance

**Comment by comment-migrator** [comment]
```header
Originally posted by Hamed Farag on 2015-09-21 08:15
```

is it work with Hijri Date ?

**Comment by comment-migrator** [comment]
```header
Originally posted by Tim on 2016-02-01 17:45
```

Thanks! Works like a charm.

**Comment by comment-migrator** [comment]
```header
Originally posted by Shivendra Narayan on 2016-10-26 00:29
```

Hi  
I was trying the date format var format = "yy-mm-dd"; to var format = "mm-dd-yy";  
However once I change it in html fine calendar control renders correctly but while search it throws error in search result webpart. Any idea how to fix this?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2016-10-27 11:19
```

This probably has something to do with the format in which you pass it to search. This has to be in "yyyy-mm-dd" format otherwise it will result in an error. What you have to do is retrieve the date value you selected and format it to the correct search format. Once that is done, you can pass it through.

**Comment by comment-migrator** [comment]
```header
Originally posted by Sandhaaland on 2017-03-14 15:54
```

Hi Elio!

I tried this in in an Office365 tenant today.  
It failed with : $(...).datepicker is not a function (OnPostRender: )

I tried to remove loading of jquery, since it was already loaded.  
But still it failed.

I have added a RefinableDate managed property, and then applied the refiner to this field.

Any tip?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-03-15 15:41
```

When you are already loading jQuery, you have to be careful with loading jQuery UI. The script must be loading after jQuery.

Also check if the jQuery UI file is loading. You can check this under the network tab of your browser developer tools.

**Comment by comment-migrator** [comment]
```header
Originally posted by Mat Tias on 2017-05-20 13:15
```

Hi Elio  
could you please explain the steps on how to deploy this to SharePoint Online ?  
Thank you in advance !  
Cheers  
Mattias

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-05-22 12:08
```

\- Copy the contents from the display template from GitHub to a local HTML file.  
\- Upload the file to your SharePoint master pages gallery.  
\- Publish the file, check that the corresponding JS file is created.  
\- Configure your refinement web part.

**Comment by comment-migrator** [comment]
```header
Originally posted by Anis Ahmed on 2017-07-13 07:29
```

Hi Elio,

I must say that that the templates prepared by your works absolutely great for me and I have a learnt a lot from your articles.

I really loved the date picker refiner. I have a requirement in which we want the default results to be displayed only for the last one year. Later if users want results beyond one year then they can go to old results from the date range refiner. Can you please suggest me any piece of code to make this possible?

Thanks  
Nez

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-08-08 10:58
```

Thanks for the feedback.

You could use the "this year" token in the search query (like: LastModifiedTime:"this year"). The downside of this is that it will also limit to show older results because of the query, and you can only filter results for the current year.

Another approach could be to set the filter via JavaScript (from the refiner template) once you loaded the page.

**Reply by comment-migrator** [reply]
```header
Originally posted by Anis Ahmed on 2017-08-09 12:27
```

Thanks for your reply.

As you rightly said the query rule approach will limit the results and wont give me the option to see results beyond one year.

I tried another option of adding the following line in a new refiner template.  
AjaxControlToolkit.SliderRefinementControl.FindAndSetSliderHandlesAndValuesToRange(ctx.RefinementControl.containerId, 1, 5, this);

The problem is that the best bets (promoted results) disappears when the page loads.

Thanks  
Anis

**Comment by comment-migrator** [comment]
```header
Originally posted by Maximiliano Dicola on 2019-12-16 20:37
```

Hi Elio,

Thank you very much for your work, the refiner is excellent! However I have a problem, when i do a search it shows me 1 single document, but the truth is that there are 3 documents that fall within that criteria.  
![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/8ebee803369112bc9c689fc4033d05429366a4b53698d4d636aae988d8be46cd.jpg)

Thanks!

**Comment by comment-migrator** [comment]
```header
Originally posted by Bob on 2021-06-29 08:11
```

Hi,  
I am a big fan of your blog.  
I would like to know if is there a way to localize the date picker? My site isn't in English.

**Reply by comment-migrator** [reply]
```header
Originally posted by Bob on 2021-06-30 10:14
```

and one more question...  
Is is possible to set the datepicker range based on managed property values?

**Comment by DihiaDiya** [comment]
Hello
I would like to change the date format from : "yy-mm-dd" to format : "dd-mm-yy" , what can I do please ?
Thanks in advance

**Reply by estruyf** [reply]
It has been a long time since I've touched this, but from what I can see, is that you can change line 11 (`var format = "yy-mm-dd";`) to the format you want.

**Reply by DihiaDiya** [reply]
I have done this. I changed var format = "dd-mm-yyyy" but while search ; it returns error in search result webpart

**Reply by estruyf** [reply]
It might indeed only work to display, but not to search, as the format needs to be correct for the search query.

**Reply by DihiaDiya** [reply]
Thank you for your help
